src/network/networkd-wwan-bus.c

```diff
@@ -0,0 +1,1285 @@
+                                "ip-type",       "u",
+                                "allow-roaming", "b",
+                                "pin",           "s",
+                                "operator-id",   "s"
+                               );
+
+        if (!key)
+                return NULL;
+
+        STRV_FOREACH_PAIR(prop, type, SIMPLE_PROP_TYPES)
+                if (streq_ptr(*prop, key))
+                        return *type;
+
+        return NULL;
+}
+
+static int sd_bus_call_method_async_props(
+                sd_bus *bus,
+                sd_bus_slot **slot,
+                const char *destination,
+                const char *path,
+                const char *interface,
+                const char *member,
+                sd_bus_message_handler_t callback,
+                void *userdata,
+                Link *link) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        int r;
+
+        assert_return(bus, -EINVAL);
+        assert_return(bus = bus_resolve(bus), -ENOPKG);
+        assert_return(!bus_origin_changed(bus), -ECHILD);
+
+        if (!BUS_IS_OPEN(bus->state))
+                return -ENOTCONN;
+
+        r = sd_bus_message_new_method_call(bus, &m, destination, path, interface, member);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_open_container(m, 'a', "{sv}");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        STRV_FOREACH(prop, link->network->modem_simple_connect_props) {
+                const char *type;
+                _cleanup_free_ char *left = NULL;
+                _cleanup_free_ char *right = NULL;
+                signed int right_i;
+                unsigned int right_u;
+
+                r = split_pair(*prop, "=", &left, &right);
+                type = prop_type_lookup(left);
+                if ((r < 0) || !type) {
+                        log_error("ModemManager: malformed simple connect option: %s, file: %s",
+                                  *prop, link->network->filename);
+                        return -EINVAL;
+                }
+
+                if (streq_ptr(left, "ip-type")) {
+                        MMBearerIpFamily ip_type = prop_iptype_lookup(right);
+
+                        r = sd_bus_message_append(m, "{sv}", left, type, (uint32_t)ip_type);
+                        continue;
+                }
+
+                if (streq_ptr(left, "allowed-auth")) {
+                        MMBearerAllowedAuth auth = prop_auth_lookup(right);
+
+                        r = sd_bus_message_append(m, "{sv}", left, type, (uint32_t)auth);
+                        continue;
+                }
+
+                switch (type[0]) {
+                case SD_BUS_TYPE_BOOLEAN:
+                        r = parse_boolean(right);
+                        if (r < 0)
+                                return -EINVAL;
+                        r = sd_bus_message_append(m, "{sv}", left, type, (bool)r);
```

Comment:
No, please use `int` for boolean. This is an API that supports old C (hence without `bool`).